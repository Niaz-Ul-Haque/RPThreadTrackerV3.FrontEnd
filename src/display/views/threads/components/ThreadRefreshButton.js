import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'reactstrap';

const propTypes = {
	isArchive: PropTypes.bool.isRequired,
	refreshThreads: PropTypes.func.isRequired
};

const ThreadRefreshButton = (props) => {
	const { refreshThreads, isArchive } = props;
	return (
		<div className="thread-refresh-button">
			<Button
				color="primary"
				onClick={() => refreshThreads(isArchive)}
			>
				Refresh <i className="fas fa-sync-alt" />
			</Button>
		</div>
	);
};
ThreadRefreshButton.propTypes = propTypes;
export default ThreadRefreshButton;
