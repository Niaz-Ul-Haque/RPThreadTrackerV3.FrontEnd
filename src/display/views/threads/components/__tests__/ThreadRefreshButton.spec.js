// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '~/utility/helpers.unit';
import ThreadRefreshButton from '../ThreadRefreshButton';
// #endregion imports

const createTestProps = propOverrides => ({
	isArchive: true,
	refreshThreads: jest.fn(),
	...propOverrides
});
describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const props = createTestProps();
			const jsx = (<ThreadRefreshButton {...props} />);
			const element = shallow(jsx);
			expect(element).toMatchSnapshot();
		});
	});
});

describe('behavior', () => {
	describe('refreshThreads', () => {
		it('should trigger refresh property function on click', () => {
			const refreshThreads = jest.fn();
			const props = createTestProps({ refreshThreads });
			const jsx = (<ThreadRefreshButton {...props} />);
			const element = shallow(jsx);
			const button = getSpecWrapper(element, 'thread-refresh-button-submit');
			button.simulate('click');
			expect(refreshThreads).toHaveBeenCalledTimes(1);
			expect(refreshThreads).toHaveBeenLastCalledWith(true);
		});
	});
});
