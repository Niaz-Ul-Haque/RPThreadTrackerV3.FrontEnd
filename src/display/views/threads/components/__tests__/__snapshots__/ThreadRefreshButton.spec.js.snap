// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<div
  className="thread-refresh-button"
>
  <Button
    color="primary"
    data-spec="thread-refresh-button-submit"
    onClick={[Function]}
    tag="button"
  >
    Refresh 
    <i
      className="fas fa-sync-alt"
    />
  </Button>
</div>
`;
